src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php

```diff
@@ -165,13 +178,75 @@ protected function loadCommands(array $config, ContainerBuilder $container)
             $container
                 ->register(
                     'elcodi.sitemap_dumper.' . $profileName,
-                    '%elcodi.core.sitemap.command.dump_sitemap.class%'
+                    '%elcodi.sitemap_dumper.class%'
                 )
                 ->addArgument(new Reference($profile['render']))
                 ->addArgument(new Reference('elcodi.sitemap_profile.' . $profileName))
+                ->addArgument(new Reference('elcodi.sitemap_event_dispatcher'))
+                ->setPublic(true)
+                ->addTag('elcodi.sitemap_dumper');
+        }
+
+        return $this;
+    }
+
+    /**
+     * Load dumper chain
+     *
+     * @param array            $config    Configuration
+     * @param ContainerBuilder $container Container
+     *
+     * @return $this self Object
+     */
+    protected function loadDumperChain(array $config, ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('elcodi.sitemap_dumper_chain')) {
+
+            return $this;
+        }
+
+        $sitemapDumperChain = $container->getDefinition('elcodi.sitemap_dumper_chain');
+
+        $sitemapDumpers = $container->findTaggedServiceIds(
+            'elcodi.sitemap_dumper'
+        );
+
+        foreach ($sitemapDumpers as $sitemapDumperId => $tags) {
+
+            $sitemapDumperChain->addMethodCall(
+                'addSitemapDumper',
+                array(new Reference($sitemapDumperId))
+            );
+        }
+
+        return $this;
+    }
+
+    /**
+     * Load commands
+     *
+     * @param array            $config    Configuration
+     * @param ContainerBuilder $container Container
+     *
+     * @return $this self Object
+     */
+    protected function loadCommands(array $config, ContainerBuilder $container)
+    {
+        $profiles = $config['profiles'];
+
+        foreach ($profiles as $profileName => $profile) {
+
+            $container
+                ->register(
+                    'elcodi.sitemap_command.' . $profileName,
+                    '%elcodi.core.sitemap.command.dump_sitemap.class%'
+                )
+                ->addArgument(new Reference('elcodi.sitemap_dumper.' . $profileName))
```

Comment:
It is SNCRedis-style. Its much easier for final users to have an specific command with an specific action. If you work with parameters its more "complicated"